lib/portlayer/attach/communication/connector.go

```diff
@@ -199,94 +261,124 @@ func (c *Connector) processIncoming(conn net.Conn) {
 		HostKeyCallback: callback,
 	}
 
-	log.Debugf("Initiating ssh handshake with new connection attempt")
-	var (
-		ccon    ssh.Conn
-		newchan <-chan ssh.NewChannel
-		request <-chan *ssh.Request
-	)
-
-	ccon, newchan, request, err = ssh.NewClientConn(conn, "", config)
+	// create the SSH connection
+	clientConn, chans, reqs, err := ssh.NewClientConn(conn, "", config)
 	if err != nil {
 		log.Errorf("SSH connection could not be established: %s", errors.ErrorStack(err))
 		return
 	}
 
-	client := ssh.NewClient(ccon, newchan, request)
-
-	var ids []string
-	ids, err = SSHls(client)
+	// ask the IDs
+	ids, err := ContainerIDs(clientConn)
```

Comment:
I think this should be below the registration of the mux handlers - I'm uncertain whether it's significant in this case but I'm very wary of potentially making a call that expects a reply without having the mux's running - I seem to recall that leads to potential deadlocks.